src/p_pspr.c

```diff
@@ -1203,6 +1205,21 @@ void P_MovePsprites(player_t *player)
     }
   }
 
+  static spritenum_t oldsprite = -1;
+  static int oldframe = -1;
+  static fixed_t oldsxf = -1, oldsyf = -1;
+
+  const spritenum_t sprite = psp->state->sprite;
```

Comment:
Clang-tidy complains that `psp->state` could be `NULL`. Above, we have the check `if (psp->sprite) ...`, should we add a check here too?

Reply:
Done in f3dfa680fc90c413f0f20acaa8d114187068ca98.